cli/src/semgrep/run_scan.py

```diff
@@ -114,15 +122,93 @@ def get_file_ignore() -> FileIgnore:
+def print_product_status(sast_enabled: bool = True, sca_enabled: bool = False) -> None:
+    """
+    (Simple) print the statuses of enabled products to stdout when the user
+    is given the product-focused CLI UX treatment.
+    """
+    learn_more_url = with_color(
+        Colors.cyan, "https://semgrep.dev/products/cloud-platform/", underline=True
+    )
+    login_command = with_color(Colors.gray, "`semgrep login`")
+    is_logged_in = auth.get_token() is not None
+    all_enabled = True  # assume all enabled until we find a disabled product
+
+    sections = [
+        (
+            "Semgrep OSS",
+            True,
+            [
+                "Basic security coverage for first-party code vulnerabilities.",
+            ],
+        ),
+        (
+            "Semgrep Code (SAST)",
+            is_logged_in and sast_enabled,
+            [
+                "Find and fix vulnerabilities in the code you write with advanced scanning and expert security rules.",
+            ],
+        ),
+        (
+            "Semgrep Supply Chain (SCA)",
+            sca_enabled,
+            [
+                "Find and fix the reachable vulnerabilities in your OSS dependencies.",
+            ],
+        ),
+    ]
+
+    for name, enabled, features in sections:
+        all_enabled = all_enabled and enabled
+        console.print(
+            f"\n{with_feature_status(enabled=enabled)} {with_color(Colors.foreground, name, bold=True)}"
+        )
+        for feature in features:
+            console.print(f"  {with_feature_status(enabled=enabled)} {feature}")
+
+    if not is_logged_in:
+        message = "\n".join(
+            wrap(
+                f"💎 Get started with all Semgrep products via {login_command}.",
+                width=80,
+            )
+            + [f"✨ Learn more at {learn_more_url}."]
+        )
+        console.print(f"\n{message}\n")
+    elif not all_enabled:
+        # TODO: Handle cases where SAST / SCA are not enabled (GROW-53)
+        # We should suggest a resolution such as enabling supply chain in SCP settings, and
+        # run then `semgrep ci`. However, there are some additional edge cases to consider
+        # such as feature availability / required plan upgrades, and thus we will punt showing
+        # a resolution for now.
+        console.print(" ")  # space intentional for progress bar padding
+    else:
+        console.print(" ")  # space intentional for progress bar padding
+
+
+def print_scan_plan_header(
+    target_manager: TargetManager,
+    sast_plan: Plan,
+    sca_plan: Plan,
+    cli_ux: DesignTreatment = DesignTreatment.LEGACY,
 ) -> None:
+    """
+    Prints the number of files intended to be scanned and (optionally)
+    the number of rules to be run based on the current configuration.
+    """
     file_count = len(target_manager.get_all_files())
-    new_cli_ux = get_state().env.with_new_cli_ux
+    legacy_cli_ux = cli_ux == DesignTreatment.LEGACY
```

Comment:
I prefer an enum to those is_xxx because we could rewrite the code using a switch
so it's easier to see the different treatment.
But maybe here we need those if anyway because we want to keep the legacy code for the tests so maybe it's ok